Python/optimizer_bytecodes.c

```diff
@@ -428,6 +428,19 @@ dummy_func(void) {
             res = sym_new_type(ctx, &PyBool_Type);
             sym_set_type(value, &PyUnicode_Type);
         }
+        if (!sym_is_const(value)) {
+            assert(sym_matches_type(value, &PyUnicode_Type));
+            int next_opcode = (this_instr + 1)->opcode;
+            assert(next_opcode == _CHECK_VALIDITY_AND_SET_IP);
+            next_opcode = (this_instr + 2)->opcode;
+            // If the next uop is a guard, we can narrow value. However, we
+            // *can't* narrow res, since that would cause the guard to be
+            // removed and the narrowed value to be invalid:
+            if (next_opcode == _GUARD_IS_FALSE_POP) {
+                sym_set_const(value, Py_GetConstant(Py_CONSTANT_EMPTY_STR));
+                res = sym_new_type(ctx, &PyUnicode_Type);
```

Comment:
`res` is the output, which is always a `bool`:

```suggestion
                res = sym_new_type(ctx, &PyBool_Type);
```

Reply:
Don't forget to regenerate after!

Reply:
Fixed!